src/appleseed/foundation/math/fresnel.h

```diff
@@ -383,6 +434,138 @@ void normal_reflectance_dielectric(
     normal_reflectance *= normal_reflectance;
 }
 
+template <typename SpectrumType, typename T>
+inline void fresnel_reflectance_conductor(
+    SpectrumType&           reflectance,
+    const SpectrumType&     n,
+    const SpectrumType&     k,
+    const T                 cos_theta_i)
+{
+    //
+    //      (n^2 + k^2) cos_theta^2 - 2n cos_theta + 1
+    // Rp = ------------------------------------------
+    //      (n^2 + k^2) cos_theta^2 + 2n cos_theta + 1
+    //
+
+    //
+    //      (n^2 + k^2) - 2n cos_theta + cos_theta^2
+    // Rs = ----------------------------------------
+    //      (n^2 + k^2) + 2n cos_theta + cos_theta^2
+    //
+
+    //               Rp + Rs
+    // reflectance = -------
+    //                  2
+
+    typedef typename impl::GetValueType<SpectrumType>::ValueType ValueType;
+
+    const T cos_theta = std::abs(cos_theta_i);
+    const T cos_theta2 = square(cos_theta);
+
+    const SpectrumType n2 = n * n;
+    const SpectrumType k2 = k * k;
+    const SpectrumType n2_plus_k2 = n2 + k2;
+    const SpectrumType n_2cos_theta = n * ValueType(2.0 * cos_theta);
+
+    // Parallel.
+    const SpectrumType n2_plus_k2_cos_theta2 = n2_plus_k2 * ValueType(cos_theta2);
+    const SpectrumType one(1.0);
+    reflectance =
+        (n2_plus_k2_cos_theta2 - n_2cos_theta + one) /
+        (n2_plus_k2_cos_theta2 + n_2cos_theta + one);
+
+    // Perpendicular.
+    const SpectrumType cos_t(cos_theta);
+    reflectance +=
+        (n2_plus_k2 - n_2cos_theta + cos_t) /
+        (n2_plus_k2 + n_2cos_theta + cos_t);
+
+    reflectance *= ValueType(0.5);
+}
+
+template <typename SpectrumType>
+inline void artist_friendly_fresnel_conductor_reparameterization(
+    const SpectrumType&     normal_reflectance,
+    const SpectrumType&     edge_tint,
+    SpectrumType&           n,
+    SpectrumType&           k)
+{
+    typedef typename impl::GetValueType<SpectrumType>::ValueType ValueType;
+
+    const SpectrumType one(1.0);
+    const SpectrumType r = clamp(normal_reflectance, ValueType(0.0), ValueType(0.99));
+    const SpectrumType sqrt_r = sqrt(r);
```

Comment:
I need to use argument dependent lookup here to pick sqrt depending on SpectrumType.
std::sqrt when SpectrumType is double (in unit tests) and renderer::sqrt when 
SpectrumType is a DynamicSpectrum (when rendering).